egs2/TEMPLATE/asr1/asr.sh

```diff
@@ -184,6 +189,8 @@ if [ "${feats_type}" = raw ]; then
     data_feats=${dumpdir}/raw
 elif [ "${feats_type}" = fbank_pitch ]; then
     data_feats=${dumpdir}/fbank_pitch
+    # export for HOW2 as stage 2 is skipped
+    export _DUMPDIR=${data_feats}
```

Comment:
I think this is not good. A little bit tricky.

This pattern can be generalized as using data directory which has been already prepared.

How about this?

```bash
if  [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
    ...
    elif [ "${feats_type}" = extracted ]; then
        log "stage 2: Copy datadir: data/ -> ${data_feats}/"

        for dset in "${train_set}" "${dev_set}" ${eval_sets}; do
            # 1. Copy datadir
            utils/copy_data_dir.sh data/"${dset}" "${data_feats}/${dset}"

            # 2. Derive the feature dimension
            pyscripts/feats/feat-to-shape.py "scp:head -n 1 ${data_feats}/${dset}/feats.scp |" - | \
                awk '{ print $2 }' | cut -d, -f2 > ${data_feats}/${dset}/feats_dim
            echo "${feats_type}" > "${data_feats}/${dset}/feats_type"
        done
```

Reply:
I kept `skip_stage` option in `asr.sh` as I thought it could be useful with or without modifications.

Reply:
Can you delete `skip_stage`?
Actually it's rare case to use such option and it's enough to use `stage` and `stop_stage` for rare case.
  
```bash
./run.sh --stage 1 --stop-stage 3 
./run.sh --stage 5 --stop-stage 10
```

If really we need such skip techniques, it should also support to skip multiple stages, like --skip-stage 2,4,5.
However, basically, we should keep bash script simple for beginners, though already it is complex.

Reply:
> If really we need such skip techniques, it should also support to skip multiple stages, like --skip-stage 2,4,5.

I totally agree on that! Original options handled multiple stages but I only pushed for one stage for complexity purpose, I wanted you opinion on that!

For now I deleted skip_stage as requested, if you want I can add skip stage option with multiple stage support later.

Reply:
It's o.k. for now, thanks!